lib/shopify_api/bulk/query.ex

```diff
@@ -97,6 +98,47 @@ defmodule ShopifyAPI.Bulk.Query do
     end
   end
 
+  @doc """
+  Streams the HTTP GET of the passed in URL for JSONL decoding.
+
+  Warning: Since HTTPoison spawns a seperate process which uses send/receive
+  to stream HTTP fetches becareful where you use this.
+  """
+  @spec stream_fetch(String.t() | {:ok, String.t()} | {:error, any()}) :: Enumerable.t()
+  def stream_fetch({:ok, url}), do: stream_fetch(url)
+  def stream_fetch({:error, _} = error), do: error
+
+  def stream_fetch(url) do
+    url
+    |> httpoison_streamed_get()
+    # Our http chunks are not guaranteed to be on the line break, this splits up our
+    # returned chunks by line break.
+    |> Stream.transform("", fn element, acc ->
+      element = acc <> element
+
+      cond do
+        # A nicely formatted jsonl line(s)
+        # {\"test\":\"bar\"}\n{\"test\":\"baz\"}\n"
+        String.ends_with?(element, "\n") ->
+          {String.split(element, "\n", trim: true), ""}
+
+        # Contains a jsonl line and then some
+        # {\"test\":\"bar\"}\n{\"test\"
+        String.contains?(element, "\n") ->
+          {rem, e} =
+            element
+            |> String.split("\n", trim: true)
+            |> List.pop_at(-1)
+
+          {e, rem}
+
+        # No complete jsonl document yet
+        true ->
+          {[], element}
+      end
```

Comment:
This might make sense to extract to a private function. `transform_chunks_to_jsonl/2`?